File: Diseno de muros concreto/Diseño de muros concreto V2/Serializador.vb
Imports System.IO
Imports System.Runtime.Serialization.Formatters.Binary

Public Class Serializador

    Public Sub New(ByVal Ruta As String, ByVal Guardar As Boolean, ByVal Lista_serializadas As Listas_serializadas)

        If Guardar = True Then
            Serializar(Ruta, Lista_serializadas)
        Else
            Deserializar(Ruta, Lista_serializadas)
        End If

    End Sub

    Public Shared Sub Serializar(ByRef ruta As String, ByVal Lista_serializadas As Listas_serializadas)

        Dim Formatter As BinaryFormatter = New BinaryFormatter

        If ruta = "" Then
            Dim Myfile As SaveFileDialog = New SaveFileDialog
            With Myfile
                .Title = "Información Muros"
                .Filter = "Guardar Archivo |*.dmc"
            End With

            Myfile.ShowDialog()
            ruta = Myfile.FileName
        End If

        Dim Escritor As Stream = New FileStream(ruta, FileMode.Create, FileAccess.Write, FileShare.None)
        Formatter.Serialize(Escritor, Lista_serializadas)
        Escritor.Close()

    End Sub

    Public Shared Sub Deserializar(ByVal Ruta As String, ByRef Lista_Serializadas As Listas_serializadas)

        Dim Formatter As BinaryFormatter = New BinaryFormatter

        Dim Myfile As OpenFileDialog = New OpenFileDialog

        With Myfile
            .Title = "Información Muros"
            .Filter = "Guardar Archivo |*.dmc"
        End With

        Myfile.ShowDialog()
        Ruta_archivo = Myfile.FileName

        Dim Lector As Stream = New FileStream(Ruta_archivo, FileMode.Open, FileAccess.Read, FileShare.None)
        Lista_Serializadas = CType(Formatter.Deserialize(Lector), Listas_serializadas)

        Lector.Close()
    End Sub

End Class

File: Analisis muros concreto fc/Diseno muros concreto fc/Serializador.cs
using Diseño_de_muros_concreto_V2;
using System;
using System.Collections;
using System.IO;
using System.Runtime.Serialization.Formatters.Binary;
using System.Windows.Forms;

namespace Diseno_muros_concreto_fc
{
    internal class Serializador
    {
        public static void Serializar(ref string ruta, Listas_Serializadas_i Lista_i)
        {
            BinaryFormatter formatter = new BinaryFormatter();

            if (ruta == "")
            {
                SaveFileDialog Myfile = new SaveFileDialog
                {
                    Title = "Información Muros",
                    Filter = "Guardar Archivo |*.dmc"
                };

                Myfile.ShowDialog();
                ruta = Myfile.FileName;
            }

            Stream Escritor = new FileStream(ruta, FileMode.Create, FileAccess.Write, FileShare.None);
            formatter.Serialize(Escritor, Lista_i);
            Escritor.Close();
        }

        public static void Deserializar(ref string Ruta_archivo, ref Listas_Serializadas_i Lista_i)
        {
         
            BinaryFormatter Formatter = new BinaryFormatter();
            OpenFileDialog Myfile = new OpenFileDialog
            {
                Filter = "Archivo de muros|*.dmc",
                Title = "Abrir archivo"
            };

            Myfile.ShowDialog();
            Ruta_archivo = Myfile.FileName;

            if (Ruta_archivo != "")
            {
                Stream Lector = new FileStream(Ruta_archivo, FileMode.Open, FileAccess.Read, FileShare.None);
            var aux = Formatter.Deserialize(Lector);
            Lector.Close();

            if (aux.GetType().Namespace == "Diseño_de_muros_concreto_V2")
            {
                var aux2 = (Listas_serializadas)aux;
                Lista_i = new Listas_Serializadas_i
                {
                    Lista_Muros = aux2.Lista_Muros,
                    Lista_Alzados = aux2.Lista_Alzados,
                    lista_refuerzo = aux2.lista_refuerzo,
                    Muros_generales = aux2.Muros_generales,
                    Capacidad_proyecto = aux2.Capacidad_proyecto
                };
            }
            else
            {
                Lista_i = (Listas_Serializadas_i)aux;
            }

            Convertir_Listas(Lista_i);
            }
        }

        private static void Convertir_Listas(Listas_Serializadas_i Lista_i)
        {
            foreach (Muros_Consolidados_1 prueba in (IEnumerable)Lista_i.Lista_Muros)
            {
                Listas_Programa.Muros_Consolidados_Listos.Add(prueba);
            }

            foreach (Muro prueba in (IEnumerable)Lista_i.Muros_generales)
            {
                Listas_Programa.Lista_Muros.Add(prueba);

                foreach (Shells_Prop prueba2 in prueba.Shells_Muro)
                {
                    Listas_Programa.Lista_shells.Add(prueba2);
                }
            }
        }
    }
}